### PR TITLE
feat(jobs): add blockchain hash verification badges for job inputs/outputs

### PR DESCRIPTION
**Pull request by francisluz**
### 🎯 Summary
  Implements visual blockchain hash verification badges for job inputs and outputs, allowing users to see at a glance whether their job data has been verified on the blockchain.

  ### 🔄 Changes

  #### New Features
  - **Verification Badges**: Added visual indicators (✅/❌) showing blockchain hash verification status for job inputs and outputs
  - **Tooltip Support**: Interactive tooltips linking to MIP-004 documentation for user education
  - **Accessibility**: Full ARIA label support for screen readers

  #### Code Refactoring
  - **Extracted Hash Logic**: Moved hash verification from `job.service.ts` to reusable utility functions in `lib/utils/index.ts`
  - **Backward Compatibility**: Maintains support for deprecated hash formats to ensure existing jobs continue to work
  - **Improved Types**: Changed return types from empty strings to `null` for better type safety

  #### Testing
  - **Comprehensive Coverage**: Added 25 unit tests covering all new utility functions
  - **Edge Cases**: Tests handle empty data, null fields, special characters, and case sensitivity
  - **Deprecation Testing**: Validates both current and deprecated hash format handling

  #### Performance
  - **Optimized Filtering**: Post-query filtering in job repository with clear documentation explaining the approach
  - **Memoization**: Uses `useMemo` in React component to prevent unnecessary recalculations

  ### 📋 Technical Details

  **Modified Files:**
  - `src/components/jobs/job-details/job-verification-badge.tsx` - New verification badge component
  - `src/lib/utils/index.ts` - New utility functions: `getMatchedHash`, `toJobInputData`, `tryParseJson`
  - `src/lib/db/repositories/job.repository.ts` - Added post-query filtering for verified jobs
  - `src/lib/services/job.service.ts` - Removed duplicate hash verification logic
  - `src/components/accordion-wrapper.tsx` - Added support for verification badges
  - `src/components/jobs/job-details/job-details.tsx` - Integrated badges into job details view

  **New Test File:**
  - `src/lib/utils/__tests__/index.test.ts` - Comprehensive test suite for utility functions

  ### ✅ Testing
  - All 65 tests passing
  - No ESLint warnings or errors
  - Build verification successful

  ### 🔒 Security
  - Safe JSON parsing with error handling
  - No exposure of sensitive data
  - Input validation on all utility functions

  ### 📸 Screenshots
  _Verification badges appear next to Input/Output section headers in the job details view, showing green checkmarks for verified hashes and red X marks for unverified ones._
  
image

  ### 🔗 Related
  - Implements MIP-004 hash verification standards
  - Documentation: https://docs.masumi.network/mips/_mip-004

  ### ✨ Impact
  - **User Experience**: Users can now instantly see if their job data is blockchain-verified
  - **Code Quality**: Cleaner separation of concerns with reusable utilities
  - **Maintainability**: Well-tested code with comprehensive documentation
  - **Performance**: Efficient verification with proper memoization
  
  Generate by Claude code.

### CODE REVIEW

File: web-app/messages/en.json [hunk 1]
```diff
@@ -282,6 +282,11 @@
             }
           }
         },
+        "VerificationBadge": {
+          "tooltip": "{direction}-Hash verified on blockchain. Click here to learn more",
+          "verifiedAria": "{direction}-Hash verified",
+          "unverifiedAria": "{direction}-Hash not verified"
```
[the remark this answers is not in the document]
Comment (francisluz) [reply web-app/messages/en.json:288]: I'll leave the `Click here to learn more` as its not obvious that the user can click on a tooltip generally.

File: web-app/src/components/jobs/job-details/job-verification-badge.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+          href={siteConfig.links.mip004Docs}
+          target="_blank"
+          rel="noreferrer noopener"
+        >
+          {t("VerificationBadge.tooltip", { direction: directionText })}
+        </Link>
+      </TooltipContent>
+    </Tooltip>
+  );
+}
```
[the remark this answers is not in the document]
Comment (francisluz) [reply web-app/src/components/jobs/job-details/job-verification-badge.tsx:104]: done

File: web-app/src/lib/db/repositories/job.repository.ts [hunk 3]
```diff
@@ -92,7 +96,27 @@ export const jobRepository = {
...
+      const matched = getMatchedHash(
+        "output",
+        outputObj,
+        job.identifierFromPurchaser,
+        job.outputHash,
+      );
+      // Keep jobs where hash verification failed (not yet verified on blockchain)
+      return matched === null;
+    });
+    return filtered.map(mapJobWithStatus);
```
[the remark this answers is not in the document]
Comment (mrgrauel) [reply web-app/src/lib/db/repositories/job.repository.ts:119]: > Stop syncing the agent status if the output hashes matches

That doesn't make sense because the user could still be unsatisfied with the result and request a refund.

I started a conversation on Linear on this topic https://linear.app/nmkr/issue/DEV-130/verify-input-and-output-hashes#comment-3d41265c

File: web-app/src/components/jobs/job-details/job-verification-badge.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+        </span>
+      </TooltipTrigger>
+      <TooltipContent>
+        <Link
+          href={siteConfig.links.mip004Docs}
+          target="_blank"
+          rel="noreferrer noopener"
+        >
+          {t("VerificationBadge.tooltip", { direction: directionText })}
+        </Link>
```
[the remark this answers is not in the document]
Comment (mrgrauel) [reply web-app/src/components/jobs/job-details/job-verification-badge.tsx]: Thanks, I didn't see that. I was just reading the acceptance criteria 🤦

File: web-app/src/lib/db/repositories/job.repository.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (mrgrauel) [reply web-app/src/lib/db/repositories/job.repository.ts:119]: @francisluz it can and should be removed. Patrick only wanted the agent sync from being stopped. This can be implement, but it's not requirement in my current point of view. If we don't just remove it from the requirements in the issue.

**Review by mrgrauel — CHANGES_REQUESTED** [review]
Please undo the sync changes. It was misunderstood by us.
